### DISCUSSION TITLE
Change menu (navigation bar) toggle direction to right replacing dropdown list.

### DISCUSSION THREAD

**Discussion by emersonfelipesp** [netbox-community/netbox#9629] [Ideas]
With the increasing amount of Netbox features being developed together with the new plugins coming day after day, it is becoming kind of hard to navigate through system when you are a "heavy user" of Netbox.

There are already issues going on like **[Toggle the visibility of navigation menu items #9070](https://github.com/netbox-community/netbox/issues/9070)** and **[Allow plugins to inject top-level navigation menu items #9071](https://github.com/netbox-community/netbox/issues/9071)** that certainly will help.

But I would like to suggest changing how navigation menu works.
I will use **Google's ISP Portal** as an example. See below:

### Part 1 - Always on the page (fixed).
![image](https://user-images.githubusercontent.com/24397251/176508274-a1a485bf-ca9b-410e-8652-b7569097861d.png)

### Part 2 - The button expands to show its name after hovering one of the icons.
![image](https://user-images.githubusercontent.com/24397251/176506046-0815104e-4fdf-417a-b14b-78fb7079efcb.png)

### Part 3 - The second-level options (child of the options above) appears after hovering too.
![image](https://user-images.githubusercontent.com/24397251/176506164-9d47fdc9-6ca9-4d88-b017-90dbb6017f42.png)

### Available space on the viewport (display)
It does not occupies the viewport (display) of the screen when not using the menu.
Today Netbox's menu is always there, displaying it when hovering or clicking is better in my opinion.

### Dropdown list gets endless with Netbox own features together with plugins installed.
Sometimes it is kind of boring having to scroll through menu to find some item on plugins section, as an example.
Using this right direction approach, even having a lot of buttons, it is more "readable".
- In other words: it is scalable, while dropdown lists gets harder to use as times goes by.

**Comment by candlerb** [comment]
> It does not occupies the viewport (display) of the screen when not using the menu.

I don't understand how part 1 / part 2 of what you propose is different from today. Here's what Netbox looks like to me:

![image](https://user-images.githubusercontent.com/44789/176621230-b1be7e2f-6ae9-425e-838f-3fadb7fccb6c.png)

That looks just like what you showed.  Furthermore, the menu pops out when I hover over one of these icons. If that's not what you see, then you need to deselect the "pin" icon next to the Netbox logo, to allow the menu to collapse:

![image](https://user-images.githubusercontent.com/44789/176621412-0d2f302d-83a3-45c8-9f55-c1ff11a75d86.png)

(This assumes you're using a reasonably recent version of Netbox. The whole UI was redesigned in 3.0.0, but I can't remember if the menu pinning/unpinning was available from day one).

The only difference I can see is your part 3.  Currently in Netbox, you click on a top-level menu item and it expands in the vertical direction, concertina-style, within the menu bar.  Click on a different one and that opens up, while the previous one collapses.

This works well for me, and I find easier to control than multiple levels of hovering. It does sometimes require vertical scrolling, but that could be true for a pop-up as well.

**Reply by candlerb** [reply]
Depending on your screen size, you might be affected by #8633 which made the pin vanish in certain cases.

This was fixed in [v3.1.9](https://github.com/netbox-community/netbox/releases/tag/v3.1.9).

**Reply by emersonfelipesp** [reply]
> The only difference I can see is your part 3. Currently in Netbox, you click on a top-level menu item and it expands in the vertical direction, concertina-style, within the menu bar. Click on a different one and that opens up, while the previous one collapses.

It is exactly that the main difference. The first and second is pretty equal, but then when you click some button, instead of dropdown list, it will display another column to the right.

**Comment by hagbarddenstore** [comment]
The suggestion is to expand to the right instead of expanding down as I interpret it.

**Reply by candlerb** [reply]
But the OP also made this comment about their suggestion:

-----
### Available space on the viewport (display)
It does not occupies the viewport (display) of the screen when not using the menu.
Today Netbox's menu is always there, displaying it when hovering or clicking is better in my opinion.

**Comment by jeremystretch** [comment]
Maybe it would help if you produced a proof of concept for people to try out.
